### PR TITLE
Added conversion menu on right click in text field (#191)

### PR DESCRIPTION
**Pull request by oscargus**
Accessing these conversions without going through clean up can be quite useful.

### CODE REVIEW

**Comment by matthiasgeiger** [comment]
Great! Looks good to me. :+1:

**Comment by simonharrer** [comment]
Thank you for this pull request. It really looks very good. 

Could you add tests similar to the CaseChangersTest class? We try to improve the code quality in JabRef, and automated tests for the logic package is an essential part for this. 

A very minor comment: Instead of `UNICODETOLATEX ` can you please write `UNICODE_TO_LATEX`, as it makes it a little bit more readable.

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [master]
+	- Feature: Added converters from HTML and Unicode to LaTeX on right click in text fields (#191)
```
Comment (stefan-kolb) [inline CHANGELOG]: Should be spaces instead of a tabulator. I also see this in your other PRs.

**Comment by oscargus** [comment]
OK! Will add test cases and change tab to spaces in ChangeLog.

**Comment by oscargus** [comment]
I fixed the ChangeLog issue. In my opinion the tests are more related to the HTMLConverter file than the rightclick menu as that is where the conversion logic is. As such, it makes more sense to add a test next time I work on that.

**Comment by oscargus** [comment]
OK, I now realized the test thing. I thought this was primarily the right click menu, but indeed I did add the Converters class which clearly needs testing.

However, I had an issue which I think relates to that the JabRef preferences are not accessed. It seems like Globals.prefs is not activated during testing. As such I had to add a bit of logic to check for this, which I guess in general is good, but actually running JabRef I assume it will always be non-null. Is there a better way to deal with this?

**Comment by koppor** [comment]
We thought at the Wiki (https://github.com/JabRef/jabref/wiki/Code-Howtos#test-cases) that you add `Globals.prefs = JabRefPreferences.getInstance();` somewhere in your tests.

**Comment by oscargus** [comment]
Ahh, great!

**Comment by oscargus** [comment]
The characters I used in the failed test were åäö. Is this an issue with the test environments or should we avoid directly representing every character above 128? I could of course write `"\u..."`, but since this worked when running it on my computer and I had the impression that all code should be in UTF-8 (recalling some discussion a few years back regarding the ChangeLog), I thought it was OK.

**Comment by koppor** [comment]
Yes, `UTF-8` is the way to go. `jabref\src\test\java\net\sf\jabref\logic\util\ConvertersTest.java` is encoded in `Cp1252` as far as I can see. Could you please recode it?

In my Eclipse, I had to change at the preferences "General -> Workspace" "Text file econding" to `UTF-8`. I just pushed a patch to `build.gradle` to have `gradlew eclipse` generate the right encoding information for the JabRef project.

Please vote at [GRADLE-2274](https://issues.gradle.org/browse/GRADLE-2274) to have the encoding information also transformed into eclipse at `gradlew eclipse` automatically without any patch.

**Comment by koppor** [comment]
OK, the test seem to go through now. However, the CHANGELOG seems to be filled with two lines. Are you sure about the Copyright statements in the header? At least the year in `ConversionMenu.java` doesn't match. You could also add yourself explicitly.

**Comment by simonharrer** [comment]
:+1: 

I think we can merge this. Thank you for your contribution!

**Pull request merged**